### ISSUE TITLE
Particles

### ISSUE THREAD

**Issue by stefanmielke** [stefanmielke/libdragon-extensions#5]
We should have a simple particle system that users can add on their ROMs.

The system should allow simple actions, like moving and re-sizing, and also have rendering associated with them (we'll probably need to have a 2d and a 3d version of this).

**Comment by adamcate** [comment]
Yeah, there should be various types of particles. For 2d maybe one type could be a simple, solid color rectangular particle that uses rdpq_fill_rectangle, and then have another variant that uses textured rectangles, and maybe a variant that enables 2d sprite rotation through triangle commands. The user can choose based on performance
